### ISSUE TITLE
`unwrap_used` recommends using `expect` even when the `expect_used` lint is also enabled

### ISSUE THREAD

**Issue by sgued** [rust-lang/rust-clippy#9222]
### Description

# Short description

When both the `unwrap_used` and `expect_used` lints are opted in (with `#![warn(clippy::unwrap_used, clippy::expect_used)]`) the unwrap_used lint still recommend using `expect`, even though it would still cause an error.

# Example
With this: 
```rust
#![warn(clippy::unwrap_used, clippy::expect_used)]

fn main() {
    Some(3).unwrap();

    Some(3).expect("Hello wrld!");
}
```

The error message looks like:

```
    Checking playground v0.0.1 (/playground)
warning: used `unwrap()` on `an Option` value
 --> src/main.rs:4:5
  |
4 |     Some(3).unwrap();
  |     ^^^^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> src/main.rs:1:9
  |
1 | #![warn(clippy::unwrap_used, clippy::expect_used)]
  |         ^^^^^^^^^^^^^^^^^^^
  = help: if you don't want to handle the `None` case gracefully, consider using `expect()` to provide a better panic message
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unwrap_used
```

The help message recommends using `expect` though it would still raise a warning.

# Fix suggestion

I think in that case the help message should be the same that is used in `expect_used`: "if this value is an `None`, it will panic`

### Version

```text
Tested with nightly in the [playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=be2ec742b6bcc4cc9e32a3aa983e669a)

Nightly version: 1.64.0-nightly (2022-07-20 d68e7ebc38cb42b8b237)
```

### Additional Labels

@rustbot label + C-enhancement
